lightning/src/ln/peer_handler.rs

```diff
@@ -1272,4 +1284,140 @@ mod tests {
 		peers[0].timer_tick_occured();
 		assert_eq!(peers[0].peers.lock().unwrap().peers.len(), 0);
 	}
+
+	pub struct TestRoutingMessageHandler {
+		pub chan_upds_recvd: AtomicUsize,
+		pub chan_anns_recvd: AtomicUsize,
+		pub chan_anns_sent: AtomicUsize,
+	}
+
+	impl TestRoutingMessageHandler {
+		pub fn new() -> Self {
+			TestRoutingMessageHandler {
+				chan_upds_recvd: AtomicUsize::new(0),
+				chan_anns_recvd: AtomicUsize::new(0),
+				chan_anns_sent: AtomicUsize::new(0),
+			}
+		}
+
+	}
+	impl msgs::RoutingMessageHandler for TestRoutingMessageHandler {
+		fn handle_node_announcement(&self, _msg: &msgs::NodeAnnouncement) -> Result<bool, msgs::LightningError> {
+			Err(msgs::LightningError { err: "", action: msgs::ErrorAction::IgnoreError })
+		}
+		fn handle_channel_announcement(&self, _msg: &msgs::ChannelAnnouncement) -> Result<bool, msgs::LightningError> {
+			self.chan_anns_recvd.fetch_add(1, Ordering::AcqRel);
+			Err(msgs::LightningError { err: "", action: msgs::ErrorAction::IgnoreError })
+		}
+		fn handle_channel_update(&self, _msg: &msgs::ChannelUpdate) -> Result<bool, msgs::LightningError> {
+			self.chan_upds_recvd.fetch_add(1, Ordering::AcqRel);
+			Err(msgs::LightningError { err: "", action: msgs::ErrorAction::IgnoreError })
+		}
+		fn handle_htlc_fail_channel_update(&self, _update: &msgs::HTLCFailChannelUpdate) {}
+		fn get_next_channel_announcements(&self, starting_point: u64, batch_amount: u8) -> Vec<(msgs::ChannelAnnouncement, msgs::ChannelUpdate,msgs::ChannelUpdate)> {
+			let mut chan_anns = Vec::new();
+			let batch_amt = batch_amount as u64;
+			const TOTAL_ANNS: u64 = 100;
+			let num_anns_sent = self.chan_anns_sent.load(Ordering::Acquire) as u64;
+			let end: u64 = if starting_point > 0 { min(starting_point + batch_amt, TOTAL_ANNS - num_anns_sent) } else { batch_amount.into() };
```

Comment:
You can drop the if now (and once you do that it would probably be more readable to inline the "as u64" stuff.